### PR TITLE
feat(recipes): add Qwen3.5 and Nemotron 3 Nano GB200 pretraining

### PR DESCRIPTION
**Pull request by cuichenx**
## Summary

- add library Qwen3.5 text pretraining recipes for dense 9B and 35B-A3B on 8x GB200
- add the validated Nemotron 3 Nano BF16 pretraining recipe for 8x GB200
- export stable recipe aliases for direct users and NeMo-CI selection
- preserve library recipe behavior; no `perf_recipes` or performance-harness files are added

## Configuration

Qwen3.5-9B uses `Qwen/Qwen3.5-9B-Base`, TP1/PP1/CP1/EP1, GBS128/MBS2, sequence length 4,096, native fused cross entropy, BF16 gradient reduction, scoped Transformer Engine graphs over attention/MLP, and disabled loss/gradient NaN checks as validated by the full convergence run.

Qwen3.5-35B-A3B uses the corresponding text-only MoE provider with TP1/PP1/CP1/EP8, learned routing, HybridEP, MBS1, canonical flex-dispatcher SMS configuration, FP32 router computations, and scoped TE graphs.

Nemotron 3 Nano uses TP1/PP1/CP1/EP8, GBS512/MBS2, sequence length 4,096, HybridEP, native fused cross entropy, BF16 gradient reduction, and scoped TE graphs over attention, Mamba, router, and MoE preprocessing modules.

## Validation

- all-file pre-commit passes on current `main`
- focused Qwen/Nano recipe tests, including the public Qwen text alias and exact 48k runtime overrides: 92 passed
- exhaustive offline recipe-factory matrix: 252 passed
- fresh independent `origin/main...HEAD` review; the one FP32-router finding was fixed in `84dd8aea3`, followed by re-review with no remaining actionable findings
- Qwen3.5-9B physical 8x GB200 validation: 100/100 RP2 steps, LM loss `13.24286` -> `6.583383`, zero skipped/NaN iterations ([W&B `hvdq6oa3`](https://wandb.ai/nvidia/megatron-bridge-ci-long-convergence/runs/hvdq6oa3))
- Qwen3.5-9B physical 64x GB200 convergence: 48,000/48,000 steps, 100.663296B RP2 tokens, final checkpoint saved, final LM loss `2.094276`, zero skipped/NaN iterations ([W&B `xbp3sunp`](https://wandb.ai/nvidia/megatron-bridge-ci-long-convergence/runs/xbp3sunp))
- Nemotron 3 Nano physical 64x GB200 convergence: 48,000/48,000 steps, 100.663296B RP2 tokens, final checkpoint saved, final LM loss `2.027169`, zero skipped/NaN iterations ([W&B `avbr31km`](https://wandb.ai/nvidia/megatron-bridge-ci-long-convergence/runs/avbr31km))
- Qwen3.5-35B-A3B physical 8x GB200 validation: 100/100 RP2 steps, LM loss `12.83251` -> `6.099396`, zero skipped/NaN iterations ([W&B `5w1agqzm`](https://wandb.ai/nvidia/megatron-bridge-ci-long-convergence/runs/5w1agqzm))

### CODE REVIEW

File: src/megatron/bridge/recipes/qwen/qwen35.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ruff: noqa: F401
+"""Compatibility alias for the canonical Qwen3.5 text recipe."""
+
+from __future__ import annotations
+
+from megatron.bridge.recipes.qwen.gb200.qwen35 import (
```
Comment (yaoyu-33) [inline src/megatron/bridge/recipes/qwen/qwen35.py:19]: these recipes should have text in name

File: src/megatron/bridge/recipes/qwen/gb200/qwen35.py [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    # needs it to select the registered causal-LM bridge instead of the VLM.
+    text_config.architectures = [architecture]
+    return AutoBridge.from_hf_config(text_config).to_megatron_provider(load_weights=False)
+
+
+def qwen35_text_9b_pretrain_8gpu_gb200_bf16_config() -> ConfigContainer:
+    """Return a text-only Qwen3.5-9B pretraining config for eight GB200 GPUs."""
+    cfg = _pretrain_common()
+
+    cfg.model = _qwen35_text_provider(_QWEN35_9B_BASE, "Qwen3_5ForCausalLM")
```
Comment (yaoyu-33) [inline src/megatron/bridge/recipes/qwen/gb200/qwen35.py]: _qwen35_text_provider not needed, just call directly?

File: src/megatron/bridge/recipes/nemotronh/gb200/nemotron_3_nano.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    The recipe retains the established optimizer, scheduler, routing, and BF16
+    contracts. It applies the validated GB200 TP1/EP8 HybridEP topology and
+    uses a 4,096-token sequence length for the paired NeMo-CI convergence
+    workload.
+
+    Returns:
+        GB200 BF16 pretraining configuration.
+    """
+    cfg = nemotron_3_nano_pretrain_config()
```
Comment (yaoyu-33) [inline src/megatron/bridge/recipes/nemotronh/gb200/nemotron_3_nano.py]: dont start from nemotron_3_nano_pretrain_config, start from pretrain commont and use autobridge. be flatten

**Comment by cuichenx** [comment]
Added the convergence-parity fix in 80d8c4b88: the Nemotron 3 Nano GB200 pretraining recipe now defers model vocabulary sizing to the runtime tokenizer, restoring the 32K RP2 tokenizer behavior instead of inheriting the 131072-entry HF checkpoint vocabulary. Validation: focused recipe regression test passed in the 26.08 RC2 container; all-file pre-commit passed.

**Review by yaoyu-33 — APPROVED** [review]
